src/server/master_server/metadata_manager.h

```diff
@@ -0,0 +1,74 @@
+#ifndef SERVER_MASTER_SERVER_METADATA_MANAGER_H
+#define SERVER_MASTER_SERVER_METADATA_MANAGER_H
+
+#include "absl/container/flat_hash_set.h"
+#include "src/server/master_server/lock_manager.h"
+#include "src/protos/metadata.pb.h"
+
+using protos::FileMetadata;
+
+namespace gfs {
+namespace server {
+
+/* The MetadataManager manages the following resources that are central to the 
+ * master node in GFS. 
+ * 1) the assignment of chunk handle, which is a UUID. 
+ * 2) the mapping between a file name to the chunk handles associated with it.
+ * 3) the set of deleted chunk handles. 
+ *
+ * The MetadataManager provides the following thread-safe methods: 
+ * 1) create a (default) file metadata for a given file. This involves also the
+ *    creation of underlying locks associated with this file. 
+ * 2) get the file metadata for a given file
+ * 3) create a file chunk handle
+ * 4) delete a file, this involves the deletion of all chunk handles assigned to 
+ *    this file.
+ * 5) assigning new chunk handle, this provides a unique UUID each time when the
+ *    function is called.  */
+
+class MetadataManager {
```

Comment:
How do you plan to support (1) chunk handle -> list of locations, (2) chunk handle & chunk version?

Reply:
These are essential pieces. I am going over the code in ChunkManager and are considering which is the appropriate place. Will address this soon.

Reply:
Add a TODO to remind us that this is to be determined?